Test/UnitTest1.cs

```diff
@@ -0,0 +1,42 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for interns at Metamation, Batch- July 2023.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// UnitTest1.cs
+// Test case for Expression Evaluator.
+// ------------------------------------------------------------------------------------------------
+using Eval;
+namespace TestTraining;
+
+[TestClass]
+public class TestTraining {
+   [TestMethod]
+   public void TestParse () {
+      Evaluator eval = new ();
+      // Expressions with their expected results
+      Dictionary<string, double> exp = new () {
+       { "-10 ^ 2", 100 }, { "a = 4", 4 }, { "b = 3.5", 3.5 }, { "a + b", 7.5 }, { "asin sin 90", 90 },
+       { "atan tan 45", 45 }, { "sqrt 25", 5 }, { "log 1", 0 }, { "-2 -2", -4 }, {"10/2+3", 8},
+       {"(+10+3)*2", 26}, {"(a+2) * a", 24 }, {"cos 0", 1 }, {"exp 2", 7.3890560989 }, {"cos acos 0", 0},
+       { "10 -2 -2", 6 }, { "---5", -5 }, {"-5+10", 5 }, { "-2-4", -6 }, { "---4+5--2+3", 6 },
+       { "(2+3)*-4", -20 }, { "(2+3)*+5", 25 }, { "-4*(3+5)", -32 }, { "-4+5--8", 9 }, { "-4+5-(-8)", 9 },
+       { "10^(-4+2)", .01 }, { "-10-10^2", -110 }, { "---4+5--6-2", 5 }, {"sin 45", 0.7071067812 },

```

Comment:
Instead of hardcoding the actual values, use C# to determine the correct values.